R/autoplot.pop_data.R

```diff
@@ -6,68 +6,142 @@
 
-  plot1 =
-    object %>%
-    ggplot2::ggplot(aes(x = .data$value)) +
-    ggplot2::theme_minimal() +
-    ggplot2::facet_wrap(~antigen_iso, nrow = 3)
+  # age-scatter plotting function
+  age_scatter = function(
+    object,
+    strata = NULL
+  )
+  {
+    # create basic plotting features
+    options(scipen = 999)
+    plot1 =
+      object %>%
+      ggplot2::ggplot(aes(x = .data$age,y = .data$value)) +
+      ggplot2::theme_linedraw()
+
+    if(is.null(strata))
+    {
+      plot1 = plot1 +
+        ggplot2::geom_point(size=.6, alpha=.7) +
+        ggplot2::geom_smooth(method="lm", se=FALSE) +
+        ggplot2::scale_y_log10() +
+        ggplot2::theme_linedraw() +
+        ggplot2::labs(
+          title = "Quantitative Antibody Responses by Age",
+          x = "Age",
+          y = "Value"
+        )
+    } else
+    {
+      plot1 = plot1 +
+        ggplot2::geom_point(size=.6,
+                            alpha=.7,
+                            aes(color = get(strata))) +
+        ggplot2::geom_smooth(method=lm,
+                             se=FALSE,
+                             aes(color = get(strata))
+        ) +
+        ggplot2::scale_y_log10() +
+        ggplot2::labs(
+          title = "Quantitative Antibody Responses by Age",
+          x = "Age",
+          y = "Value",
+          colour = strata
+        )
+    }
+
+  return(plot1)
+  }
+
+  # density plotting function
+  density_plot = function(
+    object,
+    log = FALSE,
+    strata = NULL
+  )
 
-  if(!is.null(strata))
   {
-    plot1 = plot1 +
-      ggplot2::geom_density(
-        ggplot2::aes(fill = get(strata)),
-        alpha = .6,
-        color = "black")
-  } else
+    plot1 =
+      object %>%
+      ggplot2::ggplot(aes(x = .data$value)) +
+      ggplot2::theme_linedraw() +
+      ggplot2::facet_wrap(~antigen_iso, nrow = 3)
+
+    if(is.null(strata))
+    {
+      plot1 = plot1 +
+        ggplot2::geom_density(alpha = .6,
+                              color = "black")
+    } else
+    {
+      plot1 = plot1 +
+        ggplot2::geom_density(
+          aes(fill = get(strata)),
+          alpha = .6,
+          color = "black") +
+        ggplot2::labs(fill = strata)
+    }
+    if(log)
+    {
+      plot1 = plot1 +
+        ggplot2::scale_x_log10() +
+        ggplot2::labs(
+          title = "Distribution of Cross-sectional Antibody Responses (Log transformed)",
+          x = "Log10(Antibody Response Value)",
+          y = "Frequency"
+        )
+    } else
+    {
+      plot1 = plot1 +
+        ggplot2::labs(
+          title = "Distribution of Cross-sectional Antibody Responses",
+          x = "Antibody Response Value",
+          y = "Frequency"
+        )
+    }
+    return(plot1)
+  }
+
+  # type error function
+  err_type = function()
   {
-    plot1 = plot1 +
-      ggplot2::geom_density(
-        # aes(fill = get(strata)),
-        alpha = .6,
-        color = "black")
-}
+    return("Error:Select the correct type: {'density','age-scatter'}")
```

Comment:
It doesn't look like this subfunction actually throws an error; please use `cli::cli_abort()` instead of `return()` (see https://cli.r-lib.org/reference/cli_abort.html and https://style.tidyverse.org/error-messages.html for details). 

Also, does this error message really need to be a subfunction? Generally, I only decompose out functions that contain more than one command.

Reply:
cli now implemented to throw an error when neither of the two options are provided.